synapseclient/client.py

```diff
@@ -204,6 +209,7 @@ def __init__(self, repoEndpoint=None, authEndpoint=None, fileHandleEndpoint=None
         self.table_query_max_sleep = 20
         self.table_query_timeout = 600  # in seconds
         self.multi_threaded = False  # if set to True, multi threaded download will be used for http and https URLs
```

Comment:
having `multi_threaded` and `max_threads` in a case where `multi_threaded` only affects download is confusing. I would be ok with removing `max_threads` completely for the next release, and defaulting to multi_threaded implementation for download as well.

Reply:
I agree it's a bit confusing to have both values. I do think it's useful to let the user customize the value, and think it is cleaner if it applies to both upload and download. We just added the multiThreaded property so I don't think we want to immediately yank it though, but I think the doc you put in suggests it may be pulled in a future release. We could make it a deprecated property however.